src/Connector.sol

```diff
@@ -238,6 +323,16 @@ contract CentrifugeConnector {
         memberlist.updateMember(user, validUntil);
     }
 
+    function handleTransfer(uint128 currency, address recipient, uint128 amount) public onlyGateway {
+        address currencyAddress = currencyIdToAddress[currency];
+        require(currencyAddress != address(0), "CentrifugeConnector/unknown-currency");
+
+        require(
```

Comment:
Just leaving this here as a comment while I'm digging into why this assumption is wrong

Reply:
Had to re-read this a few times, but I think the issue is as follows: The `escrow.approve` you are doing in the test is only possible because the escrow contract was deployed by the test contract, hence the test contract (which is the `msg.sender` of this `escrow.approve` call) is a `ward` on the escrow, see https://github.com/centrifuge/connectors/blob/94885c02662e4b6695f204e1409b473389a15aa3/src/Escrow.sol#L20. But in practice, no one else would be able to call this `escrow.approve`, except for the `Connector` contract, which will be a `ward` on the escrow.

So my suggestion is that within `Connector.handleTransfer`, this line should be added:
```
EscrowLike(escrow).approve(currencyAddress, address(this), amount);
```

Reply:
Thanks for looking into this! If `Escrow` is an immutable contract, how would this solution work if we deploy a new version of `Connector`? Would the old version of Connector need to add the new Connector address to the `wards` of the escrow?

I guess that when we deploy the Escrow, we first file the first Connector version as a ward and when/if we deploy a new version of Connector, we `deny` the former and `rely` on the latter?

Reply:
I was just thinking about this as well :) In the deploy scripts, when deploying the escrow, the `msg.sender` will become a ward. Then, within that same deploy script, we should immediately `rely` some admin origin, and `deny` the `msg.sender`. So in the future, the admin origin can manage the wards on the escrow. And at some point we can add logic to Connectors for Centrifuge Chain governance to pass messages to manage this, and the admin origin becomes the backup.

Reply:
Perfect, that's a good setup. Will make these changes 👍

Reply:
Made the changes the contract and the tests, left the deploy script changes to a separate issue: https://github.com/centrifuge/connectors/issues/62